### ISSUE TITLE
UTF-8 Normalization

### ISSUE THREAD

**Issue by SamuelYvon** [gunnarmorling/1brc#351]
Hey!

Can we assume the input is UTF-8 normalized? Many SIMD-powered implementation, or implementations that do not rely on building Strings right away will assume the input is UTF-8 normalized. Is this a valid assumption?

**Comment by sharpobject** [comment]
Yes, you can assume that byte-wise different strings are different.

Because normalization is about equivalence of sequences of code-points and because the reference solution does not contain any code to normalize the city names, I think that the reference solution treats equivalent names represented by different sequences of code-points as unequal.
